python/looker_sdk/rtl/model.py

```diff
@@ -34,6 +43,124 @@ class Model:
     """Base model for all generated models.
     """
 
+    def _key_to_attr(self, key):
+        """Appends the trailing _ to python reserved words.
+        """
+        if key[-1] == "_":
+            raise KeyError(key)
+        if key in keyword.kwlist:
+            key = f"{key}_"
+        return key
+
+    def __getitem__(self, key):
+        key = self._key_to_attr(key)
+        try:
+            ret = getattr(self, key)
+        except AttributeError:
+            raise KeyError(key)
+
+        if isinstance(ret, enum.Enum):
+            ret = ret.value
+        if ret is None:
+            raise KeyError(key)
+        return ret
+
+    def __setitem__(self, key, value):
+        key = self._key_to_attr(key)
+        if not hasattr(self, key):
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute '{key}'"
+            )
+        annotation = self.__annotations__[key]
+        if isinstance(annotation, ForwardRef):
+            actual_type = eval(
+                annotation.__forward_arg__, self.__global_context, locals()
+            )
+            if isinstance(actual_type, enum.EnumMeta):
+
+                # untyped because mypy really doesn't like this enum internals stuff
+                def err(val):
+                    valid = []
+                    for v in actual_type.__members__.values():
```

Comment:
could this be a class static collection so it's not looped every time?

Reply:
python enums don't behave like regular classes so you can't really just add a static/class property - it just defines another enum member:

```
>>> class e(enum.Enum):
...     a = "a"
...     b = "b"
...     _values = ["a", "b"]

>>> e.a
<e.a: 'a'>

>>> e._values
<e._values: ['a', 'b']>

>>> e._values.value
['a', 'b']
```

we could do that (and access `actual_type._values.value`) but that sort of pollutes the enum... but we're already polluting it with `invalid_api_enum_value` so maybe that's ok?

If your concern is performance: the code inside this nested function is compiled but doesn't run unless called which only happens if the user tries to set in an invalid enum value. Also, our enums don't tend to be too big... I guess I favor this looping over polluting the codegen'd enums

Reply:
I think you made the right call